src/app/encoder.cpp

```diff
@@ -74,6 +74,59 @@ using namespace chip;
+                buf.PutLE64(static_cast<uint64_t>(value));                                                                         \
+                break;                                                                                                             \
+            case 0x28:                                                                                                             \
+                buf.Put(static_cast<uint8_t>(value));                                                                              \
+                break;                                                                                                             \
+            case 0x29:                                                                                                             \
+                buf.PutLE16(static_cast<uint16_t>(value));                                                                         \
+                break;                                                                                                             \
+            case 0x2B:                                                                                                             \
+                buf.PutLE32(static_cast<uint32_t>(value));                                                                         \
+                break;                                                                                                             \
+            case 0x2f:                                                                                                             \
+                buf.PutLE64(static_cast<uint64_t>(value));                                                                         \
+                break;                                                                                                             \
+            default:                                                                                                               \
+                ChipLogError(Zcl, "Type is not supported for report attribute: '0x%02x'", attr_type);                              \
+                break;                                                                                                             \
+            }                                                                                                                      \
+        }                                                                                                                          \
+    }                                                                                                                              \
+                                                                                                                                   \
+    uint16_t result = buf.Fit() && CanCastTo<uint16_t>(buf.Written()) ? static_cast<uint16_t>(buf.Written()) : 0;                  \
```

Comment:
The check for cast capability seems somewhat awkward.

Could we make buf have some sort of limitation on it? for example?

```
BufBound buf = BufBound(buffer, buf_length);  

buf.LimitLengthTo(std::numeric_limits<uint16_t>::max());
....
uint16_t result = buf.Fit() ? ... : 0;
```

or to avoid more casting, allow buffer.Written() to do the cast:

```
uint16_t result = buff.WrittenAmount<uint16_t>().OnError(0);
```
with some fictional Optional-like writtenAmount return thing that has OnError to cast to a final value.

likely not for this PR...